cl/search/tests/test_clean_docket_number_raw.py

```diff
@@ -0,0 +1,143 @@
+from django.core.management import call_command
+
+from cl.lib.redis_utils import get_redis_interface
+from cl.search.factories import CourtFactory, DocketFactory
+from cl.search.models import Docket
+from cl.tests.cases import TransactionTestCase
+
+
+class CleanDocketNumberRawCommandTests(TransactionTestCase):
```

Comment:
We should use `TestCase` if possible since `TransactionTestCase` slows tests down considerably.

```suggestion
class CleanDocketNumberRawCommandTests(TestCase):
```

I didn’t see any method call here that requires `TransactionTestCase`, but if a test starts failing you can wrap the method that uses a DB transaction with:

`with self.captureOnCommitCallbacks(execute=True):`

This ensures the transaction is completed before checking the assertions.

I just realized that you have tests using `TransactionTestCase` from the daemon PR. Sorry for not catching that earlier, but I think we can fix them in this PR.

What you’ll want to do is:

Use `TestCase` in `TestLLMCleanDocketNumberRaw` and `TestLLMCleanDocketNumberByCourt`.

To make the tests pass in `test_update_docket_number_updated`, wrap the `update_docket_number` call with:

```

 with self.captureOnCommitCallbacks(execute=True):
            update_docket_number(
                docket.id, self.new_docket_number, start_timestamp, self.r
            )
```

And in the tests from `TestLLMCleanDocketNumberByCourt`, use the same context manager when calling `clean_docket_number_by_court()`.

That should make the tests faster.

Reply:
Thank you Alberto! I thought I needed to use `TransactionTestCase` because otherwise the DB updates are not isolated within each test, causing the tests to fail. Thank you for the pointer! In what situation would `TransactionTestCase` be appropriate?